### ISSUE TITLE
error: builder for '...' failed with exit code 1;

### ISSUE THREAD

**Issue by yonacwy** [pete3n/nix-offline-iso#2]
First attempt, any suggestions? 

error: builder for '/nix/store/fzvaa8avcdbafa98jangbzp7mi5pd438-calamares-nixos-extensions-0.3.18.drv' failed with exit code 1;
       last 10 log lines:
       > applying patch /nix/store/ax9k8ajy90vi77sd0lrzkkz4bfmv654y-welcome.patch
       > patching file config/modules/welcome.conf
       > Hunk #1 succeeded at 82 (offset 30 lines).
       > Hunk #2 succeeded at 90 (offset 30 lines).
       > applying patch /nix/store/9iagx1ibxfx8b2y6d4dh9r5zv0ydhl3g-nixos.patch
       > patching file modules/nixos/main.py
       > Hunk #2 FAILED at 377.
       > Hunk #3 succeeded at 656 (offset 13 lines).
       > Hunk #4 succeeded at 735 (offset 13 lines).
       > 1 out of 4 hunks FAILED -- saving rejects to file modules/nixos/main.py.rej
       For full logs, run 'nix log /nix/store/fzvaa8avcdbafa98jangbzp7mi5pd438-calamares-nixos-extensions-0.3.18.drv'.
error: 1 dependencies of derivation '/nix/store/dwqbjx772g3b8vgqphvz2awm8sayxwqy-closure-info.drv' failed to build
error: 1 dependencies of derivation '/nix/store/fn2gy19hvfsshyn9kaq3v6knf0w4dbal-closure-info.drv' failed to build
error: 1 dependencies of derivation '/nix/store/62g7lw0q8jr0h1ac0nczwx4pakpvw57g-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/n6k84gf2z8kfkqxzr0b9hvplarpscfv4-nixos-24.05.20241209.7109b68-x86_64-linux.iso.drv' failed to build

nix log...

[student@nixos:~/nix-offline-iso-main]$ nix log /nix/store/fzvaa8avcdbafa98jangbzp7mi5pd438-calamares-nixos-extensions-0.3.18.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/fzvaa8avcdbafa98jangbzp7mi5pd438-calamares-nixos-extensions-0.3.18.drv^*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/jkw6fwabcdrsgprv0qcw3fmp7jnlgj4g-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
applying patch /nix/store/ax9k8ajy90vi77sd0lrzkkz4bfmv654y-welcome.patch
patching file config/modules/welcome.conf
Hunk #1 succeeded at 82 (offset 30 lines).
Hunk #2 succeeded at 90 (offset 30 lines).
applying patch /nix/store/9iagx1ibxfx8b2y6d4dh9r5zv0ydhl3g-nixos.patch
patching file modules/nixos/main.py
Hunk #2 FAILED at 377.
Hunk #3 succeeded at 656 (offset 13 lines).
Hunk #4 succeeded at 735 (offset 13 lines).
1 out of 4 hunks FAILED -- saving rejects to file modules/nixos/main.py.rej

**Comment by pete3n** [comment]
@yonacwy The issue is that I wrote my offline patch for calamares-nixos-extensions-0.3.14 but didn't pin that package version in my flake. Nixpkgs 23.11 is now using v0.3.18 which has some changes that are breaking my patch (looks like v0.3.15 added LUKS2 setup support). As an immediate solution, I created a v0.3.14 branch which is pinned to the old v0.3.14 version which should get you up and running. When I get a chance I will circle back and update the main branch so that it works with the current version.

For now, clone the v0.3.14 branch and try building with that. Let me know if you still have issues or if this resolves it.
